### ISSUE TITLE
Tekken 6 (ULUS10466): Fighters' legs tremble/shudder/"flicker" in battle

### ISSUE THREAD

**Issue by thedax** [hrydgard/ppsspp#5399]
I'm surprised nobody has reported this yet, but it's been afflicting this game since at least June or July of 2013, so a precise range of builds is not really possible. I haven't seen reports of this happening before then, however. @solarmystic: do you know if this problem existed back in the 0.7 or 0.7.5 days, or even earlier?

Basically what happens is both fighters' legs will move back and forth on their own in a glitchy manner when they're idle, and not being moved by the player/AI. It's especially noticable if one goes into practice mode and pushes the opponent up against the wall.

Steps for reproduction:
1. Start Tekken 6.
2. Enter practice mode.
3. Pick any two characters, and push the CPU opponent against the wall.
4. Observe the trembling/twitching/"flickering".

I tested with 0.9.6-881-g8c63f38, Windows x64. No combination of Vertex Caching, HW transform, texture scaling, buffered/unbuffered rendering, Jit on/off, Fast Memory on/off, SoftGPU, etc., fixes it.

What should I look for in the GE Debugger?

SoftGPU testing isn't an option; the game is completely broken graphically in it:
![Screenshot 01](http://i.minus.com/ibzusMeNgPZXd5.jpg)

Here's a video of the problem:
https://drive.google.com/file/d/0BwcSjM-JLvbkX2VKM1NhOTVtQmc/edit?usp=sharing

**Comment by ALOIYD** [comment]
This bug is too funny, especially when fighting Azazel, pls fix, Cannot survive more.

**Comment by justm34evah** [comment]
hrydgard I didn't actually realize you were the owner of this emulator project. I want to thank you for your hard work from the bottom of my heart! PPSSPP is a great improvement over JPCSP (tested them  both) and only on PPSSPP I had a great gaming experience. I enjoyed playing Blazblue Continuum Shift II, Tekken 6 and Fate: Unlimited Codes. I love your work. I definitely hope for a fix of the Tekken 6 issue but hrydgard should take his time to do what is best and I am pretty sure he is trying his best to fix the issue! Maybe this year we'll play a perfect version of Tekken 6 while next year Tekken 7 arrives (on PC too!)

**Comment by issuer** [comment]
Found something interesting, this bug also happens in the Phantasy Star Portable games. In the animations "(hold)", your character's legs tremble just like in Tekken 6.

**Comment by VIRGINKLM** [comment]
Possibly, when there is somekind of rounding involved and then you make a conversion (float16 to float32 or float32 to float16) this makes it a bit inaccurate, hence the animations or the positions of the legs could be a bit off.

**Comment by tabnk** [comment]
Didn't realize this bug already occur more than a year ago.

**Comment by hrydgard** [comment]
@tabnk , the legs have been shaking as long as the emulator has been able to run the game. Still no idea what the cause is.

**Comment by unknownbrackets** [comment]
Do the legs jitter even slightly differently with jit on/off or respect fpu rounding on/off?  Or is it the same jitter?

I wonder if this is related to the Ridge Racer, Castlevania, and Dissidia replay problems.  If there were all caused by some cpu bug squirreled away somewhere...

-[Unknown]

**Comment by tabnk** [comment]
Tired Jit On and Off but no different at all.

**Comment by hrydgard** [comment]
We've been though this before, JIT has no bearing on the leg shaking. It's either a slight precision issue with some instruction, a genuine bug in the interpreter (and also JIT if the buggy instruction is jitted), or some weird timing issue causing some bone matrices to get updated at the wrong time or offset somehow.

**Comment by unknownbrackets** [comment]
Well, I know that they still jitter with jit off, was just wondering if it's even slightly differently.  There are some remaining precision (and INF/NAN) differences between the interpreter and jit.

If bones are ignored, they don't jitter at all?  Do we know it's bones?

-[Unknown]

**Comment by tabnk** [comment]
No possible fix for this ?

**Comment by hrydgard** [comment]
As has been said repeatedly, nobody has figured out how to fix it yet. Asking for it again doesn't help.

**Comment by VIRGINKLM** [comment]
Also, it could be unrelated but it's worth giving a look at it. Notice the buzzing feeling on the screen (Tekken DR), like the screen is mounted to a vibration mechanism, kinda like how PCSX2 handles interlaced output. (I try to describe the effect as accurately as I can) There is a possibility trying to understand what causes that issue would help understand what's causing the limb shaking issue.

**Comment by unknownbrackets** [comment]
IIRC, the PSP LCD screen has such a low response rate that interlacing effects aren't even visible.  If the buzzing is fast enough, maybe it does actually happen on a PSP and you just can't notice?  Although from the videos it doesn't seem exactly like that...

-[Unknown]

**Comment by VIRGINKLM** [comment]
I did 2 tests to be sure it's a PPSSPP issue:

1. Used component cables to connect to a TV so I can be sure that PSP's screen is not too cheap to reveal the issue (seemed OK to me).
2. Recorded footage at PSP Native resolution of 2 fights and transfered them on PSP to play them back so I could check if the buzzing effect is visible on PSP's screen (and turns out it is visible).

So it's definitely a PPSSPP issue.

**Comment by unknownbrackets** [comment]
Okay.  Is this game multiplayer?  Can you record a fight where you control both players?

If it were possible to get the same exact recording (e.g. character 1 standing still, character 2 immediately moving left at full speed from start, and continuing until the time runs out, or etc.), with one reproducing the problem (PPSSPP) and the other not (PSP), or especially even two of each, then it might at least be possible to see what in the recording is incorrect.  It's not certain that would lead to a fix, but it's an idea.

-[Unknown]

**Comment by justm34evah** [comment]
News regarding this issue. It seems to be related to the character's position. For example, if you choose an infinite stage, the more you move away from your starting position, the more legs will tremble/flicker. If you return to your starting position the problem barely seems noticeable. Hope it helps!

**Comment by hrydgard** [comment]
That's quite interesting, means it is probably a floating point math precision issue of some sort, as that's a typical kind of thing that can happen, as precision drops the further from the origin you are. But that kind of thing really shouldn't happen in arenas that small, heh.

**Comment by justm34evah** [comment]
It has been so long since this issue came out. No fixes as of yet? Disappointing... :( I'd love to help, but my programming skills are close to zero. Tekken 6 is my fave game on the PSP.

**Comment by justm34evah** [comment]
I've tried the latest version (1.1.1.306), the problem still occurs, but the game looks and runs great. It's kinda sad that it's the only problem...Any genius around? haha

**Comment by tabnk** [comment]
Awaiting some fix as well but not likely to happen anytime soon. Just play on REAL PSP better

**Comment by VIRGINKLM** [comment]
I wonder if a double precision hack would do anything to it, good or bad, if it does affect it in any way, not that it's the right thing to do but anyway.

**Comment by justm34evah** [comment]
Hrydgard, any opinions? Can you apply that precision hack? I want to play
the game properly and on a full screen .lol.

2015-11-18 17:49 GMT+02:00 VIRGINKLM <notifications@github.com>:

**Comment by hrydgard** [comment]
There is no "double precision hack". If I had one, I would have used it to figure out what the real problem is and fixed it. Right now I really have no idea.

**Comment by justm34evah** [comment]
Hey, hrydgard, I remember you were saying sth about a floating point math precision. Have you tried working that one out? Or...it still doesn't work. So basically, there is a starting point in a fighting arena and the further each character is from that starting point, the more visible is that leg shaking issue. It's still the same with newer builds. And I'm guessing the character's legs shake more on breakable arenas because their origin is on the ground floor...

**Comment by hrydgard** [comment]
Whatever math error this is, it may be related to the #8548 laser issue.

**Comment by justm34evah** [comment]
Great, great keep trying to find out the issue. T7 is coming soon so we
need to play T6 properly. Thank you for looking into this matter, it's been
an issue for a long time. Best luck finding the cause and solving it. Can't
wait! :)

2016-02-04 17:12 GMT+02:00 Henrik Rydgård <notifications@github.com>:

**Comment by ALOIYD** [comment]
...It's not fixed yet?

Meh, i'd better go Tekken 7 for now.

**Comment by hrydgard** [comment]
You can still play it, the legs just shake a little. Still mystified as to why.

**Comment by justm34evah** [comment]
We all want to play Tekken 7. Lol. Just pray it releases on PC otherwise
most of us will not be able to enjoy it. Oh well, it would be their loss. I
still hope Tekken 6 PSP will be perfectly emulated in the near future, it's
a nice game.

2016-02-07 19:30 GMT+02:00 ALOIYD <notifications@github.com>:

**Comment by Zetchzie** [comment]
I love tekken its my fav fighting game along with Dead or Alive.

**Comment by justm34evah** [comment]
Tekken 7 is almost here and 6 is still has this annoying visual bug. :( No revisions seem to fix the leg shaking issue.

**Comment by fuhuan416** [comment]
I tested Tekken with Xenia and Cemu, the results tell me about character's datas by GPU works.

**Comment by VIRGINKLM** [comment]
I feel the result is due to somekind of rounding issue. Legs seem to shake with progressive agression depending on you distance from the middle of the stage in all 3 axes. It affects models even when they are using somekind non-battle animation such as Jin's intro cutscene in Story Mode. Like the game does it's best to reset/fix somekind of collision error.

**Comment by justm34evah** [comment]
It's quite possibly related to this, but no one seems to care about it,
it's not like Tekken 6 is not the best 3D PSP game, and PPSSPP is not the
only way we can play T6 on PC. I mean, I understand there are other issues,
but this has been on for far too long. Oh well, I give up, not that I can
do anything about it. Thank you for your hard work though, all of you, and
thanks to PPSSPP devs we can at least play T6 at perfect speed.

On Jul 25, 2016 11:10 PM, "VIRGINKLM" <notifications@github.com> wrote:
>
> I feel the result is due to somekind of rounding issue. Legs seem to
shake with progressive agression depending on you distance from the middle
of the stage in all 3 axes. It affects models even when they are using
somekind non-battle animation such as Jin's intro cutscene in Story Mode.
Like the game does it's best to reset/fix somekind of collision error.

**Comment by ALOIYD** [comment]
Rest in peace Tekken 6

The only PSP game that got a hilarious gamebreaking bug (trembling legs push you hard) and didn't get fixed

**Comment by justm34evah** [comment]
I fell your grief, honestly. It's like fate conspired against us, Tekken
fans. I mean, I never like thought I could actually play the game on PC
seeing how it was console exclusive for PS3 and X360, I never knew it got
released for the PSP, when I found out and remembered my cousin had a PSP,
bought it and got the game. Then I dumped it when I found out you could
actually play PSP games on PC (since the JPCSP days), the framerate was
horrible, however it got better and I eventually sold my PSP as well seeing
how I didn't use it much.

PPSSPP was like the best thing ever when it came out, awesome speed, great
configuration, mobile friendly. I tried Tekken 6 on it as soon as possible
and never noticed the leg shaking issue at first, but as I played more the
problem was more noticeable. Must I say Tekken 6 on PSP has better reviews
due to better loading times and no crappy Tekken Force like Story Mode (the
story is gonna be explained in the next game anyway, it's usually this
way). And of course, it's the best 3D fighting game on the platform. The
leg shaking issue is an annoying visual bug that takes away the whole
experience and like some users said, it's hilarious. I can't train with my
character without restarting every 2 minutes due to this issue.

2016-08-08 5:14 GMT+03:00 ALOIYD <notifications@github.com>:

**Comment by VIRGINKLM** [comment]
You are being overdramatic on an issue reporting site. Chill out and give a second to the devs. They deserve it.

**Comment by justm34evah** [comment]
They do, but it's not like 2 years were not enogh, lol. Perhaps their
priorities are elsewhere. I completed Tekken 6 it a long time ago. They did
a great job with the speed and graphics, the devs, and the emulator is one
of the greatest ever up there with Dolphin, but this leg shaking issue has
been bothering me since Tekken 6 is the only game I play using PPSSPP. :)

2016-08-20 12:51 GMT+03:00 VIRGINKLM <notifications@github.com>:

**Comment by justm34evah** [comment]
Tested the game again using PPSSPP yesterday's version. No improvements
whatsoever. Is this issue so hard to fix? It would be great to make this
run perfectly before Tekken 7 hits PC in June. Honestly, the leg shaking
issue is the only issue Tekken 6 has.

2016-08-22 21:05 GMT+03:00 t0nyfightfan . <t0nyandr3w@gmail.com>:

**Comment by hrydgard** [comment]
@justm34evah Yes, it's so hard to fix. At least when a game crashes, you can find out where. But figuring out exactly what is going wrong with the math anywhere in a huge game like Tekken is not fun or easy, especially since the results are _almost_ right.

If you can figure it out, well, please do so.

**Comment by justm34evah** [comment]
I'd love to, honestly, but math was never my strong point. :P Thans for
your hard work, really, and if it's so hard then I quit. Other than testing
the game, I can't do much. Perhaps RPCSP3 or Xenia will emulate it better
in the future, who knows? I have nothing left to do but hope. See ya' and
good luck improving the emulator for the other games! It's the last time I
come up with this subject, promise.

2017-01-28 23:09 GMT+02:00 Henrik Rydgård <notifications@github.com>:

**Comment by hrydgard** [comment]
Right, Xenia is probably the best hope :) This is a very unusual type of bug. We already have extensive tests of most of the PSP VFPU's built-in math functions, and we're generally very close to the hardware so how something can cause shaking this bad still baffles me.

**Comment by paulodeleo** [comment]
As the video from the OP is long gone, I have just uploaded a high quality demo to [youtube](https://www.youtube.com/watch?v=bXLjJAbDteI) for anyone that wants to see the bug and maybe get interested in trying to fix it.

**Comment by justm34evah** [comment]
Looks like a lost cause at this point.....

**Comment by hrydgard** [comment]
That post didn't add much information to the issue.

**Comment by justm34evah** [comment]
I wonder if it Tekken 6 on PPSSPP will ever be fixed and perfectly playable.....I just want this thread alive at least.....The leg shake is basically the only issue this game has.

**Comment by tabnk** [comment]
Almost 4 years :D

**Comment by VIRGINKLM** [comment]
Whatever it is, it seems to have some kind of a constant refresh rate, like every shake happens on every even frame so it's like 15 shakes per second. Dunno if that helps narrowing down the issue with the debugger.

**Comment by justm34evah** [comment]
We also know it's a proximity issue, the farther the characters get from
the initial position, the more their legs will shake. This is why it's more
noticeable on breakable ground arenas, the center of the stage is down
below I think.

**Comment by hrydgard** [comment]
My guess is that the game tries to align the feet with the slope of the ground using some math which goes wrong somewhere. The thing where it gets worse the farther from the origin would suggest some floating point precision issue. It might be doing two operations during the alignment, one forward and the inverse of it, and they are subtly different enough for things to blow up. Or it may be a completely different problem... I mean, it could even be some sort of OS timing issue where the feet alignment code gets run at the wrong time of the frame, but seems unlikely.

We already have tests that make sure that all or most VFPU instructions work the same as on the real thing but there might be something we missed, and we know that the precision of many instructions is not perfect (vsin, vcos, vrot, mmul, vdot for example), while close.

My plan for digging into this would look something like this:

* Make a log of all FPU and VFPU instructions used by the game.
* Switch to interpreter (for ease of modification)
* Subtly sabotage all the logged instructions, one by one, to give a slightly wrong value. If one makes the shake look similar but simply worse, that might be a likely culprit. 
* Write really detailed tests for that instruction, run on a real PSP and see if we can get closer.

Step 3 may or may not reach any kind of conclusion so it is a bit of a wild goose chase...

**Comment by justm34evah** [comment]
Sounds great! Looking forward to your test results.

**Comment by hrydgard** [comment]
Didn't say I was going to do it anytime soon... Priority is stabilizing Vulkan and getting 1.5 out there.

But if anyone else want to give it a shot now, go ahead :)

**Comment by justm34evah** [comment]
That's a bit sad, but sure, you do what's necessary. Perhaps someone else
will give it a try. I for one, have no programming skills.

**Comment by leopard20** [comment]
@hrydgard
Sorry for bringing this up again. It must've given you guys quite a headache!

Just wondering if you are gonna test your above theory any time soon, since Vulkan is out and nearly perfect (at least in my opinion).

**Comment by LunaMoo** [comment]
Testing floating point accuracy might not be a complete waste of time, but I suspect wouldn't lead to fixing this game as it might as well be just a "simple" timing issue which as we know affects a lot of games, where it's not about 2 + 2 not equaling 4, but about 2 being added to 4 and equaling 6 aka the math is probably correct, it just runs some of that math at wrong time getting unexpected results. If math was wrong it would more likely affect all animations or at least all frames of same animation, but that's not the case.

 Anyway this issue is soo long and has soo many annoying bumping and requests, it doesn't seem worth wasting a weekend over staring at disassembly:P.

**Comment by Jinzombie** [comment]
@LunaMoo In Tekken 6 (ULUS10466): Fighters' legs tremble/shudder/"flicker" in battle I never experienced that since I use this cheat code that lock the fps to 30 I tried both Practice Stage Walls & No Walls and 5-10 minutes. My phone only Support OpenGL backend :)
_C1 Force 30 FPS [Fixed]
_L 0xE0030064 0x001C239C
_L 0xE002001E 0x103E4B40
_L 0x203E4B40 0x0000001E
_L 0x20022EE4 0x3D088889//2/30=0.03333
_L 0x20452890 0x00000002
_L 0x20452890 0x00000000//FPS

**Comment by justm34evah** [comment]
30 fps? Ugh.

**Comment by Jinzombie** [comment]
@justm34evah my GPU (Mali-450) is to weak to get 60 FPS that's why I use 30 FPS cheat code atleast I never experienced the flicker/shudder/tremble in fighters leg 😅

**Comment by paulodeleo** [comment]
@Jinzombie can you confirm that you can't reproduce the tremble legs bug using this 30 FPS cheat, on the same conditions as [this video](https://www.youtube.com/watch?v=bXLjJAbDteI) that I demo it (stage with no walls, tremble legs effect increase as characters get far from the initial center position and than go back to center)?

That would be a confirmation that, somehow, FPS has something related to the issue and, maybe, this information can help leading to a possible fix.

**Comment by leopard20** [comment]
This cheat code is for the Japanese version of the game though. Maybe it's different than the US and EU version. Maybe the Japanese version doesn't have this issue at all.

**Comment by justm34evah** [comment]
Omoshiroi...

**Comment by leopard20** [comment]
I tried some 30 FPS cheats on the US version and it didn't solve anything (also tried frameskipping 1, which is kinda the same as using a 30 FPS cheat). Maybe later I'll test the Japanese version (gotta download it first)

**Comment by leopard20** [comment]
Well, I tested the EU, US and JP versions and all of them had this issue.
Also, the above code doesn't work in any of them.

**Comment by leopard20** [comment]
That's the modded! Not the original version.
![screenshot_2018-04-11-20-53-39-140_org ppsspp ppsspp](https://user-images.githubusercontent.com/16115820/38629871-7d95bcb6-3dca-11e8-9ae1-a6b41a0a7828.png)
Maybe that's why it works for you.

**Comment by Jinzombie** [comment]
@Leopard20 Maybe 😅

**Comment by paulodeleo** [comment]
If the modded version unintentionally fixes the bug, I read it as some kind of progress...

**Comment by leopard20** [comment]
I watched a couple of YouTube videos of this modded version. The bug was still there.
@Jinzombie Do you think you could record a minute of the game as proof?

**Comment by TheConfuZzledDude** [comment]
Just thought I'd chime in here just to say that this isn't a purely graphical glitch, it seems to actually affect the hitboxes, which seems strange.

Video here: [https://youtu.be/Jy43rya0RNU](https://youtu.be/Jy43rya0RNU)

**Comment by leopard20** [comment]
@hrydgard Would be nice to finally fix it by 1.7.0.

**Comment by hrydgard** [comment]
Would be nice if someone could figure out what the problem is :/  

It's pretty clearly CPU emulation related though, especially if it affects the hitboxes.

**Comment by leopard20** [comment]
@hrydgard I would give it a chance but I don't really know what I should look for!
@weihuoya You seem to be pretty good at debugging! Any ideas about this issue?

**Comment by leopard20** [comment]
More findings:
The direction of the shaking changes with the fighters' direction. It's like the legs are being pulled towards the center! 
In this video, the fighters are originally aligned towards the origin. Watch what happens:
https://www.dropbox.com/s/3wj8qmfo9ts0fq4/20180607_165700.mp4?dl=0

**Comment by weihuoya** [comment]
have no idea about it.
maybe we can decompile it, and see how it works. :sweat_smile:
```
1)	Dump eboot.bin of the game using UMDGen
2)	Decrypt it if it isn’t yet using pspdecrypter
3)	Output the assembly code with: prxtool –w eboot.bin > eboot.asm
4)	Find the module name with: prxtool –m eboot.bin
5)	Output pseudo code with: pspdecompiler –c eboot.bin > eboot.c
```

**Comment by leopard20** [comment]
@weihuoya I dunno... Seems a bit impractical to me! Can you imagine reading and figuring out all that stuff?! 😵
@hrydgard What do you think?

**Comment by unknownbrackets** [comment]
Just noting here for posterity (sorry for the notifications, this isn't a fix):

This seems to potentially be related to #12082.  It was found that on ARM, vmmul retaining too much precision made the shaking of the legs worse.  A recent change in the latest git build should have reduced this, probably to the same level as Windows/Linux/Mac builds.

Please don't reply asking if this means a fix is coming: it doesn't necessarily, but it is a lead on the cause.  More will be noted in this issue if more progress is made.

-[Unknown]

**Comment by hrydgard** [comment]
We've now confirmed that the lead was correct - with more PSP-accurate emulation of vmmul and similar instructions, the leg shaking stops completely. It's a lot more complex than just the difference between the previous ARM and ARM64 implementations though, see #2990.

Still a ways to go before we can merge the new code, there's more accuracy work to be done and we need to write a faster implementation once all the edge cases have been figured out, then if it's still too slow we might only enable it for certain affected games. Or, more ideally, we might disable it only for really performance-challenged games that don't need it.

But it's looking good indeed.

**Comment by hrydgard** [comment]
This should now be fixed with https://github.com/hrydgard/ppsspp/pull/12217 merged.

**Comment by unknownbrackets** [comment]
Just a heads up for @gid15 - this is related to dot product floating point accuracy.

-[Unknown]

**Comment by hrydgard** [comment]
Was playing around with some perf stuff in https://github.com/hrydgard/ppsspp/tree/vfpu-dot-experiments when I realized that the ReadMatrix/WriteMatrix functions are very slow and might hurt more than the actual computation. Those can be special-cased and optimized which might give us enough speed back.

**Comment by hrydgard** [comment]
Some profiling shows that vfpu_dot is considerably heavier than the ReadMatrix/WriteMatrix calls, but that they indeed shouldn't be overlooked for optimization.

**Comment by tabnk** [comment]
Finally (so many years) it FIXED shaking leg issue for both Tekken Dark and Tekken 6. Importantly no much performance hit on both my PC and Android build.

**Comment by Emulatorer** [comment]
@tabnk Tekken5 doesn't have a bug in CPU emulation 🤔

**Comment by thedax** [comment]
Great to finally see this fixed. It was one of the oldest outstanding issues, IIRC.

**Comment by ssinenominee** [comment]
> Great to finally see this fixed. It was one of the oldest outstanding issues, IIRC.

please how do I fix this in layman terms, I don't understand any of the jargon used in the merged threads that fixes the issue.
Could I PLEASE get a step 1-10? *in layman terms, like what I have to do and where I have to go.
SORRY that you have to hold my hand...
Thanks a lot for helping me play PSP games guys...you rock!

**Comment by ssinenominee** [comment]
> Finally (so many years) it FIXED shaking leg issue for both Tekken Dark and Tekken 6. Importantly no much performance hit on both my PC and Android build.

please how do I fix this in layman terms, I don't understand any of the jargon used in the merged threads that fixes the issue.
Could I PLEASE get a step 1-10? *in layman terms, like what I have to do and where I have to go.
SORRY that you have to hold my hand...
Thanks a lot for helping me play PSP games guys...you rock!

**Comment by leopard20** [comment]
@ssinenominee What steps? Just download the latest build from here:
http://ppsspp.org/downloads.html

Go to the bottom of the page and download the latest development build. (latest as of now: 1.8.0-560)

**Comment by ssinenominee** [comment]
> @ssinenominee What steps? Just download the latest build from here:
> http://ppsspp.org/downloads.html
> 
> Go to the bottom of the page and download the latest development build. (latest as of now: 1.8.0-560)

ROFL....thanks a lot man, didn't know it was that easy. I appreciate it. Godspeed

**Comment by proganime1200** [comment]
still happening

**Comment by hrydgard** [comment]
@proganime1200 What version of PPSSPP are you using? Can you check the game ID of your Tekken 6 iso? This really shouldn't be happening anymore.

**Comment by HeriKuen** [comment]
Thanks I tried VFPU_MTX in the checklist position, the problem was resolved, thank you all...

**Comment by hrydgard** [comment]
I see a few mods have popped up: https://report.ppsspp.org/games?name=tekken 

Were you simply playing one of those?

**Comment by HeriKuen** [comment]
Yes Tekken 6 The beginning went well without a hitch, but in some arenas it still experiences the same bug.

**Comment by DreaMBlazeR** [comment]
Add this cheat to your PPSSPP/CHEAT/(Insert Tekken Game Id here)

![IMG_20210825_164645_462](https://user-images.githubusercontent.com/89394472/130758578-4d32d414-77b1-4f08-80a3-568b9b40b3e4.jpg)

Turn the Custom Cpu fix on then set your system emulator speed setting to 60hz

![Screenshot_2021-08-25-16-44-32](https://user-images.githubusercontent.com/89394472/130758634-0a2f246a-2a5a-412d-a754-2fedcfd121a8.png)

And those legs won't dance anymore with the stuttering "Yudeleehihooo" cause fps ia fixed. Please note you can use 120hz or higher if you use  higher resolutions.

**Comment by Gamemulatorer** [comment]
DreamBlazer No need for that it only decrease the fps capacity of the game 🤷
Shaking legs already fixed in v1.9.0

**Comment by DiegoAldaco** [comment]
Tekken 6 (ULUS10466)
